### PR TITLE
Update Policy Matcher

### PR DESCRIPTION
**Pull request by nam-jaehyun**
- update policy matcher
- upgrade protobuf version
- update test scenarios and policies

### CODE REVIEW

**Comment by nam-jaehyun** [comment]
Once this PR is merged, I'll update the PR for the SELinux enforcer on top of this.

File: KubeArmor/core/kubeArmor.go [hunk 1]
```diff
@@ -338,50 +338,53 @@ func KubeArmor() {
...
-		// == //
+			if dm.Node.NodeIP == "" {
+				for timeout := 0; timeout <= 60; timeout++ {
+					if dm.Node.NodeIP != "" {
+						break
+					}
 
-		// wait for a while
-		time.Sleep(time.Second * 1)
+					if dm.Node.NodeIP == "" && timeout == 60 {
```
Comment (nyrahul) [inline KubeArmor/core/kubeArmor.go]: We can push this condition outside of the for loop and just check for single condition `dm.Node.NodeIP == ""`.

File: KubeArmor/core/kubeArmor.go [hunk 2]
```diff
@@ -338,50 +338,53 @@ func KubeArmor() {
 		dm.Node.KernelVersion = kl.GetCommandOutputWithoutErr("uname", []string{"-r"})
 		dm.Node.KernelVersion = strings.TrimSuffix(dm.Node.KernelVersion, "\n")
 
-		dm.Node.PolicyEnabled = tp.KubeArmorPolicyEnabled
+		kg.Print("Updated the node information")
 
-		cfg.GlobalCfg.Policy = false
-		cfg.GlobalCfg.HostPolicy = true
+	} else if cfg.GlobalCfg.K8sEnv {
+		if K8s.InitK8sClient() {
```
Comment (nyrahul) [inline KubeArmor/core/kubeArmor.go]: We can reduce nesting if the other condition is handled ... 
```suggestion
		if !K8s.InitK8sClient() {
		       print error
		       return
		}
		// ... normal processing
		
```

**Review by nyrahul — CHANGES_REQUESTED** [review]

**Comment by nam-jaehyun** [comment]
Updated the code according to @nyrahul 's feedback.

**Pull request merged**